cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -342,8 +342,18 @@ std::tuple<int64_t, size_type, std::vector<row_group_info>>
 aggregate_reader_metadata::select_row_groups(
   host_span<std::vector<size_type> const> row_group_indices,
   int64_t skip_rows_opt,
-  std::optional<size_type> const& num_rows_opt) const
+  std::optional<size_type> const& num_rows_opt,
+  host_span<data_type const> output_dtypes,
+  std::optional<std::reference_wrapper<ast::expression const>> filter) const
 {
+  std::optional<std::vector<std::vector<size_type>>> filtered_row_group_indices;
+  if (filter.has_value()) {
+    filtered_row_group_indices =
+      filter_row_groups(row_group_indices, output_dtypes, filter.value());
+    if (filtered_row_group_indices.has_value())
+      row_group_indices =
+        host_span<std::vector<size_type> const>(filtered_row_group_indices.value());
```

Comment:
Always wrap the body of a conditional or loop in braces.
```suggestion
    if (filtered_row_group_indices.has_value()) {
      row_group_indices =
        host_span<std::vector<size_type> const>(filtered_row_group_indices.value());
    }
```

Reply:
added braces on all if conditions in the changes in this PR.